Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -9,15 +10,13 @@ internal static class StringExtensions
 {
     /// <summary>
     /// Finds the first index at which the <paramref name="value"/> does not match the <paramref name="expected"/>
-    /// string anymore, accounting for the specified <paramref name="stringComparison"/>.
+    /// string anymore, accounting for the specified <paramref name="comparer"/>.
     /// </summary>
-    public static int IndexOfFirstMismatch(this string value, string expected, StringComparison stringComparison)
+    public static int IndexOfFirstMismatch(this string value, string expected, IEqualityComparer<string> comparer)
     {
-        Func<char, char, bool> comparer = GetCharComparer(stringComparison);
-
         for (int index = 0; index < value.Length; index++)
         {
-            if (index >= expected.Length || !comparer(value[index], expected[index]))
+            if (index >= expected.Length || !comparer.Equals(value[index..(index + 1)], expected[index..(index + 1)]))
```

Comment:
Just for reference:
I made this optimization in #1312 but since this is only called on the failure path I'm fine with loosing it again.

Reply:
Unfortunately I see no other way, when using `IEqualityComparer<string>`, as it has no option to compare single characters.